app/bundles/LeadBundle/Entity/Lead.php

```diff
@@ -240,6 +250,20 @@ public static function loadMetadata (ORM\ClassMetadata $metadata)
             ->cascadePersist()
             ->cascadeDetach()
             ->build();
+        
+        $builder->createManyToOne('stage', 'Mautic\StageBundle\Entity\Stage')
+            ->cascadePersist()
+            ->cascadeMerge()
+            ->addJoinColumn('stage_id', 'id', false)
```

Comment:
The `false` here should be `true` to allow the field to be null when the lead is not in a stage. This allows the foreign keys to work